lib/ViewModels/TerriaViewer.js

```diff
@@ -315,6 +315,12 @@ us via email at '+that.terria.supportEmail+'.'
 
     var scene = viewer.scene;
 
+    // Set minimum zoom distance to 25 meters. Default is 1m.
+    // Zoom crashes through terrain if the rate of zoom is higher than terrain
+    // resolution rate and the height at the more detailed LOD is higher than
+    // previous LOD.
+    scene.screenSpaceCameraController.minimumZoomDistance = 25.0;
```

Comment:
The Cesium camera already has a way to deal with this problem when handling mouse input.  I think the problem is that we're just circumventing it by calling `flyToPosition`.  Can you take a look at how it does it (probably in `ScreenSpaceCameraController`) to see if we can avoid adjusting the minimum distance?